module/zfs/zfs_vfsops.c

```diff
@@ -1209,14 +1204,6 @@ zfsvfs_setup(zfsvfs_t *zfsvfs, boolean_t mounting)
 	int error;
 	boolean_t readonly = zfs_is_readonly(zfsvfs);
 
-	/*
-	 * Check for a bad on-disk format version now since we
-	 * lied about owning the dataset readonly before.
-	 */
-	if (!readonly &&
-	    dmu_objset_incompatible_encryption_version(zfsvfs->z_os))
-		return (SET_ERROR(EROFS));
```

Comment:
To be clear, this check is no longer needed since it's now handled properly in `dmu_objset_own_impl()` since we're now tell the truth about the dataset being r/w.